uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/util/SslUtil.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.util;
+
+import com.aws.greengrass.testing.mqtt5.client.MqttLib;
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.jcajce.util.DefaultJcaJceHelper;
+import org.bouncycastle.jcajce.util.JcaJceHelper;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openssl.PEMParser;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.Security;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.spec.PKCS8EncodedKeySpec;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManagerFactory;
+
+public final class SslUtil {
+    private SslUtil() {
+    }
+
+    /**
+     * generate SSL socket factory.
+     *
+     * @param connectionParams MQTT connection parameters
+     * @throws IOException on errors
+     * @throws GeneralSecurityException on errors
+     */
+    public static SSLSocketFactory getSocketFactory(MqttLib.ConnectionParams connectionParams)
+            throws IOException, GeneralSecurityException {
+        return getSocketFactory(connectionParams.getCa(), connectionParams.getCert(), connectionParams.getKey());
+    }
+
+    /**
+     * generate SSL socket factory.
+     *
+     * @param caCrtFile CA
+     * @param crtFile certification
+     * @param keyFile private key
+     * @throws IOException on errors
+     * @throws GeneralSecurityException on errors
+     */
+    @SuppressWarnings({"PMD.CloseResource", "PMD.AvoidFileStream"})
```

Comment:
Please follow PMD message and 
1. close used resources if possible
2. Do not use FileString due to other data class provided by java

Only if really can't avoid PMD warnings - use supression.

Reply:
Can resolved in future because implementation of connect() out of scope of current task.

Reply:
these remarks will be resolved in the next request.